### ISSUE TITLE
Trying to build using GitBASH

### ISSUE THREAD

**Issue by beachbouy** [spleen1981/retroarch-kodi-addon-CoreELEC#23]
[retroarch-kodi_20240210_200845.log](https://github.com/spleen1981/retroarch-kodi-addon-CoreELEC/files/14228680/retroarch-kodi_20240210_200845.log)
I'm trying to build the aarch64 version using GitBASH. I've cloned the Git according to the instructions - copy and paste the commands, but all I get is errors even with the standard build. Also When using the git clone command the lakka directory is empty so I manually populated it with the files and folders after a number of missing file errors.
What am I doing wrong? Here is the Output:

$ ARCH=aarch64 HOOK=aarch64_to_arm_userspace ./build.sh

Building RetroArch KODI add-on for CoreELEC:

GIT_SSL_NO_VERIFY set to 1
Enter version tag [e.g. v1.0.0]: 1.6.5
DISTRO=Lakka
PROJECT=Amlogic-ce
DEVICE=Amlogic-ng
ARCH=aarch64
VERSION=1.6.5

Working in: /c/retroarch-kodi-addon-CoreELEC/scripts/..
Temporary project folder: /c/retroarch-kodi-addon-CoreELEC/scripts/../retroarch_work/2024-02-10_193737

Target zip: /c/retroarch-kodi-addon-CoreELEC/scripts/../build/script.retroarch.launcher.Amlogic-ng.aarch64/script.retroarch.launcher.Amlogic-ng.aarch64-1.6.5.zip

Folder '/c/retroarch-kodi-addon-CoreELEC/scripts/../build' exists.
Folder '/c/retroarch-kodi-addon-CoreELEC/scripts/../build/script.retroarch.launcher.Amlogic-ng.aarch64' exists.
Folder '/c/retroarch-kodi-addon-CoreELEC/scripts/../build/script.retroarch.launcher.Amlogic-ng.aarch64/resources' exists.

Applying /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/01-ra_common.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/hooks/aarch64_to_arm_userspace/01-opengl-meson-coreelec.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/Amlogic-ng/91-revert_flycast.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/Amlogic-ng/02-ra_set_mali_fbdev.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/Amlogic-ng/01-AMLGX_libmali.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/96-ra_xmb_alpha_100.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/95-bump_dosbox-svn.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/94-bump_meson.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/93-bump_glib.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/92-bump_m4.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/91-bump_ra.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/90-scummvm_bump_cloud_shared_deps.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/55-pcre2_ncurses_link.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/53-ra_alsa_freeze.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/52-libcec_bump.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/51-cec-mini-kb.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/04-ra_add_reboot_shutdown.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/03-add-mupen64plus.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/02-ra_disable_cores_updater.patch (failed)
Reverting /c/retroarch-kodi-addon-CoreELEC/scripts/../patches/common/01-ra_common.patch (failed)

**Comment by spleen1981** [comment]
- You must use `git clone --recursive` to clone Lakka submodule. Copying manually the Lakka folder is not working in your case (patches do not apply) as probably you are using a different reference commit.
- `HOOK=aarch64_to_arm_userspace` is specific to compile a aarch64 build which can run in arm userspace. It was an experimental feature now outdated and not tested recently (probably it will be dropped soon). It's not what you are looking for.
- As I have not added yet a custom `Amlogic-no` device, you will need to compile giving all needed (Lakka) parameters, hence `PROJECT=Amlogic DEVICE=AMLGX ARCH=aarch64` as a minimum.

**Comment by beachbouy** [comment]
> * You must use `git clone --recursive` to clone Lakka submodule. Copying manually the Lakka folder is not working in your case (patches do not apply) as probably you are using a different reference commit.
> 
>     * `HOOK=aarch64_to_arm_userspace` is specific to compile a aarch64 build which can run in arm userspace. It was an experimental feature now outdated and not tested recently (probably it will be dropped soon). It's not what you are looking for.
> 
>     * As I have not added yet a custom `Amlogic-no` device, you will need to compile giving all needed (Lakka) parameters, hence `PROJECT=Amlogic DEVICE=AMLGX ARCH=aarch64` as a minimum.

I tried cloning recursively, but I always got errors with the lakka repo and the clone failed every time. So I cloned the lakka repo separately and it worked, thats how I got the above.

I'm trying out CoreElec NE and it seems to work on my device so I just wanted a 64 bit retroarch to go with it. NG version works ok until recently... Both versions are crashing for no apparent reason soon after boot into kodi....

**Comment by spleen1981** [comment]
> I tried cloning recursively, but I always got errors with the lakka repo and the clone failed every time.

This is a simple git command to recursively clone a github repo, what's the error you got? Anyway I'm not familiar with gitbash, you may try with WSL and regular git, it will work.

> So I cloned the lakka repo separately and it worked, thats how I got the above.

It didn't work properly, as the patches fail to apply and the building process stops. Cloning Lakka manaually should work, but make sure the repo is clean (which does not seem the case from your log, `git reset --hard` Lakka repo) and `git checkout` to the [required commit](https://github.com/spleen1981/retroarch-kodi-addon-CoreELEC/blob/33bcae60823a844ec974978d326f8076e7d65582/scripts/build_functions.sh#L90). Or just wipe everything and clone recursively.

> NG version works ok until recently... Both versions are crashing for no apparent reason soon after boot into kodi....

Maybe you can give some details and open a separate issue if needed

> I'm trying out CoreElec NE and it seems to work on my device so I just wanted a 64 bit retroarch to go with it.

That is coming...

**Comment by beachbouy** [comment]
I've been using the NG variant for over a year on my S905x4 box. It worked great -better than the Android TV firmware. Just in the last week I tried to turn it on and it booted to kodi, but soon after it just reboots. Sometimes it reboots immediately on boot up, other times it will work for a short time before rebooting. I thought it may be an addon that got corrupted or something, but even with a clean install it does the same.  I couldn't do a filesystem repair scan through SSH, so I put the card in to a linux build and did it and that let it run a little longer, but still rebooting randomly. I managed to get into your Retroarch addon and set the box to boot to it directly, and it does work better, I was able to play a game and shut it down normally. But next time it rebooted in the middle of the game.  Yesterday I tried to use it and it won't boot past the splash screen. That is why I looked at the NE version, to see if it fixed the issue, but it has the same problem.
It's all a bit strange since I haven't changed anything on the device since I first set it up...
I'll try to get some logs.
